docs/source/guides/contributor/chapters/plugins.rst

```diff
@@ -27,8 +27,65 @@ world output in the console.
           :py:data:`avocado.core.output.LOG_JOB`, which can be used to
           output to job log files when running a job.
 
+Registering configuration options (settings)
+============================================
+
+It is very common for a plugin to allow users to configure something. A plugin
+might change its behavior depending on a specific configuration option.
+
+Usually, those configurations came basically from configuration files and
```

Comment:
"Frequently, those options come from configuration files or from the command-line argument."